command/job_status.go

```diff
@@ -666,6 +718,51 @@ func (c *JobStatusCommand) outputFailedPlacements(failedEval *api.Evaluation) {
 	}
 }
 
+func createJsonJobsOutput(client *api.Client, allAllocs bool, jobIds ...string) ([]JobJson, error) {
+	jsonJobs := make([]JobJson, len(jobIds))
+
+	for i, id := range jobIds {
+		job, _, err := client.Jobs().Info(id, &api.QueryOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Error querying job info: %s", err)
+		}
+
+		var q *api.QueryOptions
+		if job.Namespace != nil {
+			q = &api.QueryOptions{Namespace: *job.Namespace}
+		}
```

Comment:
I did some testing of this branch with jobs in namespaces and discovered that we're not passing in the correct namespace for each job ID we're querying, and instead end up picking up the namespace set earlier. What that results in is wildcard namespaces being broken with the `-json` flag:

```
$ nomad job status -namespace '*'
ID      Namespace  Type     Priority  Status          Submit Date
httpd   default    service  50        dead (stopped)  2024-02-22T12:53:17-05:00
httpd   prod       service  50        running         2024-02-22T12:53:35-05:00
system  default    system   50        dead (stopped)  2024-02-22T12:53:20-05:00

$ nomad job status -json -namespace '*'
Error querying job info: Unexpected response code: 404 (job not found)
```

So I think we need to pass in namespace/ID pairs to this function.